File: src/Nexora.Domain/Entities/Notification.cs
using Nexora.Domain.Enums;

namespace Nexora.Domain.Entities;

public sealed class Notification : BaseEntity
{
    public Guid UserId { get; set; }
    public User User { get; set; } = default!;

    public string Title { get; set; } = default!;
    public string Message { get; set; } = default!;
    public NotificationType Type { get; set; }

    public bool IsRead { get; set; } = false;
    public DateTime? ReadAtUtc { get; set; }
}

File: src/Nexora.Web/src/types/notification.ts
export interface NotificationDto {
  id: string;
  userId: string;
  title: string;
  message: string;
  type: 'Order' | 'Campaign' | 'Security' | 'System' | string;
  isRead: boolean;
  readAtUtc: string | null;
  createdAtUtc: string;
}
